modules/local/metabat2.nf

```diff
@@ -0,0 +1,50 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process METABAT2 {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+
+    conda (params.enable_conda ? "bioconda::metabat2=2.15" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/metabat2:2.15--h986a166_1"
+    } else {
+        container "quay.io/biocontainers/metabat2:2.15--h986a166_1"
+    }
+
+    input:
+    tuple val(meta), path (assembly)
+    path(bam)
```

Comment:
```suggestion
    tuple val(meta), path (assembly), path(bam)
```

Reply:
Just for my understanding on the above suggestion. Why put all the inputs in a single tuple? Is it because the metadata or ID is related to both `bam` and the `assembly` ? I personally find it easier to run the tools when the inputs can be added separately.

Reply:
> Is it because the metadata or ID is related to both `bam` and the `assembly` ?

Yes, this.

Reply:
Done!